jsonschema/scripts/templates/dotnet.dotnet.erb

```diff
@@ -0,0 +1,150 @@
+<%- @schemas.each do |key, schema| -%>
+<%= class_name(key) -%>.cs
+using System;
+using System.Collections.Generic;
+
+// ------------------------------------------------------------------------------
+// This code was generated based on the Cucumber JSON schema
+// Changes to this file may cause incorrect behavior and will be lost if 
+// the code is regenerated.
+// ------------------------------------------------------------------------------
+
+namespace Io.Cucumber.Messages.Types;
+
+/**
+ * Represents the <%= class_name(key) %> message in Cucumber's message protocol
+ * @see <a href="https://github.com/cucumber/messages" >Github - Cucumber - Messages</a>
+ <%- if not (schema['description'] || []).empty? -%>
+ *
+<%= format_description(schema['description'], indent_string: '') %>
+<%- end -%>
+ */
+
+public sealed class <%= class_name(key) %> 
+{
+  <%- schema['properties'].each do |property_name, property| 
+      required = (schema['required'] || []).index(property_name) 
+      isValueType = type_for(class_name(key), property_name, property) == "long" || type_for(class_name(key), property_name, property) == "bool"
+  -%>
+    <%- if not (property['description'] || []).empty? -%>
+    /**
+    <%= format_description(property['description'], indent_string: '    ') %>
+     */
+    <%- end -%>
+    <%- if isValueType && !required -%>
+    public Nullable<<%= type_for(class_name(key), property_name, property) -%>> <%= capitalize(property_name) %> { get; private set; }
+    <%- else -%>
+    public <%= type_for(class_name(key), property_name, property) -%> <%= capitalize(property_name) %> { get; private set; }
+    <%- end -%>
+  <%- end -%>
+
+  <%- if (schema['required'] || []).empty? -%>
+  <%- schema['properties'].each_with_index do |(property_name, property), index| -%>
+
+    public static <%= class_name(key) %> Create(<%= type_for(class_name(key), property_name, property) -%> <%= property_name %>) 
+    {
+        return new <%= class_name(key) %>(
+        <%- schema['properties'].each_with_index do |(property_name_2, property_2), index_2| -%>
+          <%- if property_name_2 == property_name -%>
+            Require<<%= type_for(class_name(key), property_name, property) -%>>(<%= property_name %>, "<%= capitalize(property_name) %>", "<%= class_name(key) %>.<%= capitalize(property_name) %> cannot be null")<%= index_2 < schema['properties'].length-1 ? ',' : '' %>
+          <%- else -%>
+            null<%= index_2 < schema['properties'].length-1 ? ',' : '' %>
+          <%- end -%>
+        <%- end -%>
+        );
+    }
+  <%- end -%>
+  <%- end -%>
+
+    public <%= class_name(key) %>(
+    <%- schema['properties'].each_with_index do |(property_name, property), index| 
+        isValueType = type_for(class_name(key), property_name, property) == "long" || type_for(class_name(key), property_name, property) == "bool"
```

Comment:
If this is pushed into your class you then get the win here

Reply:
Ignore this comment @clrudolphi based on the above comment. We will tackle this after the fact once this is approved and merged.